windows/RNScreens/ScreenContainer.cpp

```diff
@@ -0,0 +1,47 @@
+#include "pch.h"
+#include "NativeModules.h"
+#include "JSValueXaml.h"
+#include "ScreenContainer.h"
+
+namespace winrt {
+    using namespace Microsoft::ReactNative;
+    using namespace Windows::Foundation;
+    using namespace Windows::Foundation::Collections;
+    using namespace Windows::UI;
+    using namespace Windows::UI::Xaml;
+    using namespace Windows::UI::Xaml::Controls;
+}
+
+namespace winrt::RNScreens::implementation {
+    ScreenContainer::ScreenContainer(winrt::Microsoft::ReactNative::IReactContext reactContext)
+        : m_reactContext(reactContext),
+          m_children({ winrt::single_threaded_vector<Windows::UI::Xaml::UIElement>() }) {
+    }
+
+    void ScreenContainer::addScreen(Screen& screen, int64_t) {
+        auto uiElement = screen.try_as<UIElement>();
+        if (!uiElement)
+            return;

```

Comment:
I guess that could be added, I decided not to throw one at the time since exceptions are not that user-friendly when working with react-native-windows. IMHO it should be checked by the JS code if possible, but it can also be added here as a last resort check.

Reply:
We check it on Android: https://github.com/software-mansion/react-native-screens/blob/b5777665392badbc3f175535f3769bbd3653f2e5/android/src/main/java/com/swmansion/rnscreens/ScreenContainerViewManager.java#L27. It can be helpful for someone adding own JS implementation for `react-native-screens`.